added support for neic

jdbrown-USGS opened the pull request:
fixes usgs/hazdev-geoserve-ws#43

- added methods for neic regions

jmfee-usgs commented:
- remove underscores from the public facing types, only keep them for table names. (`neiccatalog` and `neicresponse`)
- change the `$NEIC` parameter to `$type`;  make the parameter values `catalog` or `response`, and throw an exception if a different value is provided.
- make the method `getNEIC` protected or private.

jdbrown-USGS commented:
- Removed underscores from the public facing types
- Changed $NEIC parameter to $type
- Throws an exception if type parameter is not neic_catalog or neic_response

jdbrown-USGS commented:
- Changed $type to $table
